src/view/editor/parts/editor-equipment.tsx

```diff
@@ -49,15 +76,75 @@ export const EditorEquipment = ({ editorId }: { editorId: string }): ReactElemen
 		}
 	}, [equipment, equipments, selection]);
 
-	return equipments ? (
-		<Select
-			items={equipments}
-			background={'gray5'}
-			label={t('label.equipment', 'Equipment')}
-			onChange={onChange}
-			disabled={disabled?.meetingRoom}
-			selection={selection}
-			multiple
-		/>
+	const equipmentsAvailability = useMemo(() => {
+		if (!equipments?.length) {
+			return [];
+		}
+		return map(equipments, (_equipment) => {
+			const equipmentInList = find(attendeesAvailabilityList, ['email', _equipment.email]);
+			const isSelected = find(selection, ['email', _equipment.email]);
+
+			if (equipmentInList) {
+				const isBusyAtTimeOfEvent = getIsBusyAtTimeOfTheEvent(
+					equipmentInList,
+					start,
+					end,
+					attendeesAvailabilityList,
+					allDay
+				);
+
+				if (isBusyAtTimeOfEvent) {
+					return {
+						..._equipment,
+						email: _equipment?.email ?? _equipment?.label,
+						customComponent: (
```

Comment:
IMO the custom component can be defined in a separated component.
In this way the new component can be used for the rooms availability too, a test for the component rendering will be easier and the current component readability will be improved

Reply:
Yes, you are right, we extracted the custom component, let me know if it is better now